diff --git a/ruby/ext/google/protobuf_c/upb.h b/ruby/ext/google/protobuf_c/upb.h
index f441c89c..767c5151 100644
--- a/ruby/ext/google/protobuf_c/upb.h
+++ b/ruby/ext/google/protobuf_c/upb.h
@@ -13,6 +13,7 @@
 #include <stdarg.h>
 #include <stdbool.h>
 #include <stddef.h>
+#include <byteswap.h>
 
 #ifdef __cplusplus
 namespace upb {
@@ -8515,6 +8516,7 @@ UPB_INLINE uint64_t upb_vencode32(uint32_t val) {
   UPB_ASSERT(bytes <= 5);
   memcpy(&ret, buf, bytes);
   UPB_ASSERT(ret <= 0xffffffffffU);
+  ret = bswap_64(ret);
   return ret;
 }
 
